scripts/syncVersions.ts

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env -S node -r ts-node/register
+
+/**
+ * This scripts makes sure that all packages use the newest versions of dependencies.
+ * Without arguments, it reads all workspaces from root package.json and checks all workspaces.
+ * Pass paths to package.json files as arguments to parse only those.
+ */
+
+import fs from 'fs';
+import path from 'path';
+
+const rootDir = path.join(__dirname, '..');
+
+interface PackageJson {
+    readonly name: string;
+    readonly version: string;
+    readonly workspaces?: string[];
+    readonly dependencies?: Record<string, string>;
+    readonly devDependencies?: Record<string, string>;
+    readonly peerDependencies?: Record<string, string>;
+}
+
+type Dependencies = keyof Pick<PackageJson, 'dependencies' | 'devDependencies' | 'peerDependencies'>;
+
+async function loadPackageJson(packageJsonPath: string): Promise<PackageJson> {
+    const error = new Error(`${packageJsonPath} does not seem to be a valid package.json file`);
+
+    let packageJson: Partial<PackageJson>;
+    try {
+        packageJson = JSON.parse(await fs.promises.readFile(packageJsonPath, 'utf-8')) as Partial<PackageJson>;
+    } catch {
+        throw error;
+    }
+
+    if (!packageJson.name || !packageJson.version) {
+        throw error;
+    }
+
+    return packageJson as PackageJson;
+}
+
+async function savePackageJson(packageJsonPath: string, packageJson: PackageJson) {
+    return await fs.promises.writeFile(packageJsonPath, JSON.stringify(packageJson, null, '    ') + '\n');
+}
+
+function getWorkspacePackageJsonPaths(packageJson: PackageJson) {
+    return packageJson.workspaces?.map((workspaceDir) => path.join(rootDir, workspaceDir, 'package.json')) ?? [];
+}
+
+function updateDependency(packageJson: PackageJson, type: Dependencies, name: string, newVersion: string) {
+    const deps = packageJson[type];
+    if (!deps) {
+        return false;
+    }
+
+    const currentVersion = deps[name];
+    if (!currentVersion) {
+        return false;
+    }
+
+    if (currentVersion !== newVersion) {
+        deps[name] = newVersion;
```

Comment:
shouldn't this be `^${newVersion}` instead? Otherwise our concept with sdk-core won't work

Reply:
it is, see line 75

Reply:
ah ok. thx